src/Microsoft.SqlTools.ServiceLayer/ServerConfigurations/ServerConfigService.cs

```diff
@@ -239,5 +239,27 @@ private ConfigProperty GetSmoConfig(Server server, int configNumber)
                 throw new ServerConfigException($"Failed to get config. config number: ${configNumber}", ex);
             }
         }
+
+        public ServerConfigProperty GetConfigByName(Server server, string configName)
+        {
+            try
+            {
+                ConfigProperty serverConfig = null;
+                foreach (ConfigProperty configProperty in server.Configuration.Properties)
+                {
+                    if (configProperty.DisplayName == configName)
+                    {
+                        serverConfig = configProperty;
+                        break;
+                    }
+                }
+
+                return serverConfig != null ? ServerConfigProperty.ToServerConfigProperty(serverConfig) : null;
+            }
+            catch (Exception ex)
+            {
+                throw new ServerConfigException($"Failed to get config. config name: ${configName}", ex);
```

Comment:
Could you add the server name to this message too? Maybe something like: $"Failed to get config property '{configName}' for server '{server.Name}'." ?